From: Paul Mundt <lethal@linux-sh.org>
Date: Thu, 5 Nov 2009 13:56:50 +0900
Subject: [PATCH 004/279] sh: perf events: Fix up uninitialized variable warning.
Git-commit: 8820002c18cd3167d2800c002f13d78fa0325402
Patch-mainline: v2.6.33-rc1
References: FATE#311392, BNC#685313
Signed-off-by: Tony Jones <tonyj@suse.de>

'config' can be unintialized, and although it's not really an error, it
still manages to trigger the -Werror with certain toolchains. Initialize
it early to shut up gcc.

Signed-off-by: Paul Mundt <lethal@linux-sh.org>
Signed-off-by: Robert Richter <robert.richter@amd.com>
---
 arch/sh/kernel/perf_event.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/arch/sh/kernel/perf_event.c b/arch/sh/kernel/perf_event.c
index 4449f0a..7ff0943 100644
--- a/arch/sh/kernel/perf_event.c
+++ b/arch/sh/kernel/perf_event.c
@@ -103,7 +103,7 @@ static int __hw_perf_event_init(struct perf_event *event)
 {
 	struct perf_event_attr *attr = &event->attr;
 	struct hw_perf_event *hwc = &event->hw;
-	int config;
+	int config = -1;
 	int err;
 
 	if (!sh_pmu_initialized())
@@ -155,8 +155,6 @@ static int __hw_perf_event_init(struct perf_event *event)
 
 		config = sh_pmu->event_map(attr->config);
 		break;
-	default:
-		return -EINVAL;
 	}
 
 	if (config == -1)
-- 
1.7.3.4

